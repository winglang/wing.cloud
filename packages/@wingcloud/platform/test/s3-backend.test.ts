import s3Backend from '../src/s3-backend';
import { test, expect, describe, afterEach, beforeEach } from 'vitest'

describe('S3Backend', () => {
  let originalEnv;

  beforeEach(() => {
    originalEnv = { ...process.env };
  });

  afterEach(() => {
    process.env = originalEnv;
  });

  test('should apply config with correct values', () => {
    const config = {
      terraform: {
        backend: {}
      }
    };

    process.env = Object.assign(process.env, {
      TF_BACKEND_BUCKET: 'mock-bucket',
      TF_BACKEND_BUCKET_REGION: 'mock-region',
      TF_BACKEND_STATE_FILE: 'mock-key',
      TF_BACKEND_LOCK_TABLE: 'mock-table',
    });

    const appliedConfig = s3Backend(config);

    expect(appliedConfig).toMatchSnapshot();
  });

  test('should throw error when env vars are not set', () => {
    const config = {
      terraform: {
        backend: {}
      }
    };

    process.env = Object.assign(process.env, {
      TF_BACKEND_BUCKET: '',
      TF_BACKEND_BUCKET_REGION: '',
      TF_BACKEND_STATE_FILE: '',
      TF_BACKEND_LOCK_TABLE: '',
    });

    expect(() => s3Backend(config)).toThrow();
  });
});