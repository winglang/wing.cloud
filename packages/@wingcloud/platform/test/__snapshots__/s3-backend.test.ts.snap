// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`S3Backend > should apply config with correct values 1`] = `
{
  "terraform": {
    "backend": {
      "s3": {
        "bucket": "mock-bucket",
        "dynamodb_table": "mock-table",
        "key": "mock-key",
        "region": "mock-region",
      },
    },
  },
}
`;
